### PR TITLE
doc: Add required jobs info

### PR DESCRIPTION
**Pull request by stevenhorsman**
Add information about what required jobs are and
our initial guidelines for how jobs are eligible for being made required, or non-required

### CODE REVIEW

**Review by fidencio — APPROVED** [review]
lgtm, thanks @stevenhorsman

**Review by GabyCT — APPROVED** [review]

**Comment by stevenhorsman** [comment]
At the vPTG there seemed to be broad agreement that this approach made sense, so I'm looking for AC review here as the next step.

**Review by wainersm — APPROVED** [review]
Hi @stevenhorsman , IMHO this policy is very good defined. Thanks!

File: ci/README.md [hunk 1]
```diff
@@ -138,6 +138,63 @@ Following those examples, the community advice during the review, and even
 asking the community directly on Slack are the best ways to get your test
 accepted.
 
+## Required tests
+
+In our CI we have two catagories of jobs - required and not-required:
```
Comment (sprt) [inline ci/README.md:143]:
```suggestion
In our CI we have two categories of jobs - required and non-required:
```

File: ci/README.md [hunk 2]
```diff
@@ -138,6 +138,63 @@ Following those examples, the community advice during the review, and even
...
+
+The [Kata Containers CI Dashboard](https://kata-containers.github.io/)
+is a useful resource to check when collecting evidence of job stability.
+At time of writing it reports the last ten days of Kata CI nightly test
+results for each job. This isn't perfect as it doesn't currently capture
+results on PRs, but is a good guideline for stability.
+
+> [!NOTE]
+> Below are general guidelines about jobs being marked as
+> required/not-required, but they are subject to change and the Kata
```
Comment (sprt) [inline ci/README.md:165]:
```suggestion
> required/non-required, but they are subject to change and the Kata
```

**Review by sprt — APPROVED** [review]
LGTM! Minor nits.

File: ci/README.md (hunk 1 above)
Comment (stevenhorsman) [reply ci/README.md]: Updated, thanks for the suggestions

**Comment by stevenhorsman** [comment]
The two failing required jobs have been removed, so closing and re-opening to try and refreshing to update this.

**Pull request merged**